src/pages/components/progress-indicator/usage.mdx

```diff
@@ -45,41 +48,218 @@ set of steps, guiding toward the completion of a specified process.
+| Focus       | When a user tabs to or clicks on the progress indicator's step, the step becomes focused, indicating the user has successfully navigated to the component.                                                                                                                                                                                    |
 
-#### Indicate the current step
+### Interactions
 
-Keeping the user informed of where they currently are within the process or task
-at hand will give them a sense of control. This helps the user to know where
-they are in relation to where they have been, and what sections are to follow.
-Clear labels should accompany the progress indicator to indicate what the user
-will accomplish within each step. Keep labels between one to two words.
+By default, the progress indicator is not interactive, providing only a visual
+update of the users' progress. There is also an option for an interactive
+progress indicator, which allows users to navigate to each step.
 
-#### Validation
+#### Mouse
 
-Use validation to confirm that a previous step has been completed. If the user
-cannot proceed onto another step without first completing a task, use an
-[Inline Notification](/components/notification/code#inline-notification) to
-inform them.
+If the progress indicator is interactive, users can navigate to a step by
+clicking anywhere within the step label container.
 
 <Row>
 <Column colLg={8}>
 
-![Example of the progress indicator in a payment UI](images/progress-indicator-usage-1.png)
+![Example of an interactive progress indicator](images/progress-indicator-usage-6.png)
 
 </Column>
 </Row>
 
+#### Keyboard
+
+The first step should be selected by default. Users can navigate between steps
+by pressing `Right` or `Left` arrow keys. For additional keyboard interactions,
+see the [Accessibility](/components/progress-indicator/accessibility/) tab.
+
+### Validation
+
+When possible, use validation to confirm that a step has been completed before
+the user continues. If any entry is invalid, the progress indicator should show
+an error state. Additionally, the invalid entry should be marked with an error
+state and include an inline error message that helps the user understand the
+problem and how to fix it. This kind of inline validation (or client-side
+validation) should happen as soon as the field loses focus.
+
+If the user cannot proceed due to a server-side issue, then an
+[Inline Notification](/components/notification/code#inline-notification) should
+appear.
+
+<Row>
+<Column colLg={8}>
+
+![Example of a progress indicator using validation](images/progress-indicator-usage-7.png)
+
+<Caption>
+  Example of a progress indicator using validation.
+</Caption>
+</Column>
+</Row>
+
+## Related
+
+- [Dialogs](/patterns/dialog-pattern/)
+- [Forms](/patterns/forms-pattern/)
+- [Status indicators](/patterns/status-indicator-pattern/)
+
```

Comment:
We probably want to be more explicit about this. Like if it's good to use another component instead of the progress indicator (see the [example of TreeView](https://carbondesignsystem.com/components/tree-view/usage/#related)) or has a short definition of the component itself (see the [example of Popover](https://carbondesignsystem.com/components/tree-view/usage/#related))

Reply:
Sounds good. I added a little more of a description.